src/components/todoList.tsx

```diff
@@ -0,0 +1,100 @@
+import classNames from 'classnames';
+import { Todo } from '../types/Todo';
+
+interface Props {
+  todos: Todo[],
+  tempTodo: Todo | null,
+  idProcessed: number,
+  disableList: boolean,
+  onDelete: (todoId: number | undefined) => () => void,
+}
+
+export const TodoList: React.FC<Props> = ({
+  todos,
+  tempTodo,
+  idProcessed,
+  disableList,
+  onDelete,
+}) => {
+  return (
+    <section className="todoapp__main is-loading">
+      {todos.map(todo => (
+        <div
+          className={classNames(
+            'todo',
+            {
+              completed: todo.completed,
+            },
+          )}
+          key={todo.id}
+        >
+          <label
+            className="todo__status-label"
+          >
+            <input
+              type="checkbox"
+              className="todo__status"
+              disabled={disableList}
+            />
+          </label>
+
+          <span className="todo__title">{todo.title}</span>
+
+          <button
+            type="button"
+            className="todo__remove"
+            onClick={onDelete(todo.id)}
```

Comment:
Instead of passing a function returning another function, simply pass the function and use currying to accept the 'todo.id'.